### PR TITLE
Add receipt header

### PR DESCRIPTION
**Pull request by babygoat**

### CODE REVIEW

**Comment by taylrj** [comment]
### Problem description

I checkout this change and test with https://github.com/twreporter/membership-fe/pull/355 in local environment.

**step to reproduce**
Creating a PATCH request to `http://localhost:8080/v1/donations/prime/orders/{orderNumber}` endpoint and get status code `204`.

This is an example of request payload:

```
donor: {name: "donor name", address: "", zip_code: "", phone_number: "", national_id: ""},
address: "",
name: "donor name",
national_id: "",
phone_number: "",
zip_code: "",
is_anonymous: false,
notes: "",
receipt_header: "reciept_title",
send_receipt: "no",
to_feedback: false,
user_id: 790
```

**current behavior**
Table`pay_by_prime_donations` has a new row with empty `receipt_header`.

| id | created_at          | updated_at          | deleted_at | user_id | details                  | merchant_id        | amount | order_number                     | currency | pay_method  | status | tappay_api_status | msg     | tappay_record_status | rec_trade_id    | bank_transaction_id | auth_code | acquirer | transaction_time    | bank_transaction_start_time | bank_transaction_end_time | bank_result_code | bank_result_msg | cardholder_email          | cardholder_phone_number | cardholder_name | car
dholder_zip_code | cardholder_address | cardholder_national_id | card_info_bin_code | card_info_last_four | card_info_issuer |
card_info_funding | card_info_type | card_info_level | card_info_country | card_info_country_code | card_info_expiry_date | send_receipt | notes | is_anonymous | linepay_point | linepay_method | receipt_header |

|  6 | 2020-09-26 03:52:44 | 2020-09-26 03:53:30 | NULL       |     790 | 一般線上單筆捐款         | GlobalTesting_CTBC |    300 | twreporter-160109236424602500000 | TWD      | credit_card | paid   |                 0 | Success |                 NULL | D20200926VA7kIm | TP20200926VA7kIm    | 838804    | TW_CTBC  | 2020-09-26 03:52:45 | 2020-09-26 03:52:44         | 2020-09-26 03:52:44       |                  |                 | taylorfang@twreporter.org |                         | donor name      |                     |                    |                        | 424242             | 4242                |                  |                 0 |              1 |                 | UNITED KINGDOM    | GB                     | NULL                  | no           |       |            0 |          NULL | NULL           |                |

**expected behavior**
Table`pay_by_prime_donations` should have a new row with `receipt_header` "reciept_title".

**Comment by babygoat** [comment]
Updated.

**Comment by taylrj** [comment]
Since users may come back to edit information after they donated successfully, we need to `GET` the single donation.

Could we add [changes like this](https://github.com/taylrj/go-api/commit/dc2aa8cb2cafe18177a66652ab01c92342a73d20) for retrieving a single periodic/prime donation (/v1/donations/(prime|periodic-donations)/orders/{order})?

**Comment by babygoat** [comment]
Updated.

**Comment by taylrj** [comment]
### Problem description

Can not empty the `receipt_header` field if it had a non-empty value before.

**steps to reproduce**
1. Creating a PATCH request to http://localhost:8080/v1/donations/prime/orders/{orderNumber} endpoint with non-empty `receipt_header`.
2.Creating a PATCH request to http://localhost:8080/v1/donations/prime/orders/{orderNumber} endpoint with empty `receipt_header`. 

**current behavior**
Table `pay_by_prime_donations` has a new row with the stale non-empty value of `receipt_header`.

**expected behavior**
Table `pay_by_prime_donations` should have a new row with empty `receipt_header`.

File: models/donation.go [hunk 1]
```diff
@@ -131,4 +133,5 @@ type PeriodicDonation struct {
 	UpdatedAt     time.Time  `json:"updated_at"`
 	UserID        uint       `gorm:"type:int(10) unsigned;not null" json:"user_id"`
 	IsAnonymous   null.Bool  `gorm:"type:tinyint(1);default:0" json:"is_anonymous"`
+	ReceiptHeader string     `gorm:"type:varchar(128)" json:"receipt_header"`
```
Comment (nickhsine) [inline models/donation.go:136]: `ReceiptHeader null.String`  to update zero value (to fix https://github.com/twreporter/go-api/pull/406#issuecomment-701173390).

**Comment by babygoat** [comment]
updated.

**Comment by taylrj** [comment]
LGTM 👍

**Comment by nickhsine** [comment]
LGTM

**Pull request merged**